projects/modular_llm/get_clusters.py

```diff
@@ -0,0 +1,168 @@
+import argparse
+import logging
+import os
+
+import huggingface_hub
+import numpy as np
+from datasets import concatenate_datasets, load_dataset
+from sentence_transformers import SentenceTransformer
+from sklearn.cluster import KMeans
+from torch.utils.data import DataLoader, Dataset, Subset
+from tqdm import tqdm
+
+from mttl.datamodule.mt_seq_to_seq_module import FlanConfig, FlanModule
+from mttl.models.ranker.adapter_ranker import AdapterRankerHelper
+
+huggingface_token = os.environ.get("HF_TOKEN")
+huggingface_hub.login(token=huggingface_token)
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--batch_size", type=int, default=64)
+parser.add_argument("--embed_dim", type=int, default=768)
+
+parser.add_argument("--subsample", type=float, default=0.2)
+
+parser.add_argument("--num_clusters", type=int, default=256)
+
+parser.add_argument("--dataset", type=str, default="orca")
+
+parser.add_argument("--output_file", type=str, default="orca_cluster.json")
+
+parser.add_argument("--encoding", type=str, default="classifier")
+
+parser.add_argument(
+    "--new_dataname", type=str, default="zhan1993/flan-10k-flat-cluster-embedding"
+)
+
+parser.add_argument(
+    "--model", type=str, default="sentence-transformers/sentence-t5-xxl"
+)
+args = parser.parse_args()
+np.random.seed(42)
+
+
+def get_text_encode(text, model):
+
+    if args.encoding == "classifier":
+        return model.get_text_encode(text).cpu().detach().numpy()
+    elif args.encoding == "embedding":
+        return model.encode(text)
+
+
+if args.encoding == "classifier":
+    model = AdapterRankerHelper.get_ranker_instance(
+        ranker_model="classifier",
+        ranker_path="zhan1993/classifier_ranker_t5_v3",
+    )
+elif args.encoding == "embedding":
+    model = SentenceTransformer(args.model)
+
+# load the dataset
+
+
+def get_orca_dataset():
+
+    dataset = load_dataset("Open-Orca/OpenOrca")
+
+    # create the subsample of the dataset.
+    dataset_size = len(dataset["train"])
+    indices = list(range(dataset_size))
+    # random indices
+    np.random.shuffle(indices)
+    split = int(np.floor(args.subsample * dataset_size))
+    subset_indices = indices[:split]
+    subset_dataset = Subset(dataset["train"], subset_indices)
+
+    train_dataloader = DataLoader(
+        subset_dataset, batch_size=args.batch_size, num_workers=8
+    )
+    all_dataloader = DataLoader(
+        dataset["train"], batch_size=args.batch_size, num_workers=8
+    )
+
+    return train_dataloader, all_dataloader, dataset["train"]
+
+
+def get_flan_dataset():
+
+    flan = FlanModule(
```

Comment:
for this you don't need to use module, just `load_dataset(dataset_name)` should work